### PR TITLE
Fixed issue where you could not register or unregister falsy items

### PR DESCRIPTION
**Pull request by BorisKozo**
Fixing issue #77 

Added some unit tests for the type registry
Fixed 2 typos in existing unit tests that were found due to this change

### CODE REVIEW

File: src/backbone.syphon.typeregistry.js [hunk 1]
```diff
@@ -17,7 +17,11 @@ _.extend(TypeRegistry.prototype, {
   // found for the specified type, the default is
   // returned.
   get: function(type){
-    return this.registeredTypes[type] || this.registeredTypes['default'];
+    if (this.registeredTypes.hasOwnProperty(type)){
```
Comment (samccone) [inline src/backbone.syphon.typeregistry.js]: This will not work in IE8
https://msdn.microsoft.com/en-us/library/ie/328kyd6z%28v=vs.94%29.aspx

Comment (megawac) [inline src/backbone.syphon.typeregistry.js]: Why won't that work in IE8?

Comment (samccone) [inline src/backbone.syphon.typeregistry.js]: heh if registeredTypes is a host object it won't work :) but i guess registeredTypes will never be a host object

Comment (BorisKozo) [inline src/backbone.syphon.typeregistry.js]: It is a regular object instantiated in line 8, I can test on IE8 (I am on Windows anyway... *ducks*)

Comment (samccone) [inline src/backbone.syphon.typeregistry.js]: heheheh :) I just read a bunch about host objects, I think we are good :dancer:

Comment (BorisKozo) [inline src/backbone.syphon.typeregistry.js]: Works with no issues in IE7 mode of IE11

Comment (samccone) [inline src/backbone.syphon.typeregistry.js]: better safe than sorry right

Comment (BorisKozo) [inline src/backbone.syphon.typeregistry.js]: That is what the drill sergeant always said during weapon training :)

**Comment by rhubarbselleven** [comment]
:+1:

**Comment by rhubarbselleven** [comment]
@BorisKozo could you change this pull request to point to merge with the minor branch?

Cheers

File: src/backbone.syphon.typeregistry.js (hunk 1 above)
Comment (thejameskyle) [inline src/backbone.syphon.typeregistry.js]: It still would be good to use `_.has`.

**Pull request closed without merging**